ups_modul/inout/inout_gsheet.py

```diff
@@ -3,178 +3,235 @@
+    batch_cells(sum_tips_ratio_list, cell_list, wks)
+
+def SUM_tips(anthill, days, wks):
+    '''
+    formula SUM tips based on ratio
+    '''
+    sum_tips_list = []
+    a1_notion, a2_notion = a_notion(2,days+6,anthill.num_of_ants+1,days+6)
+    cell_list = wks.range("{}:{}".format(a1_notion, a2_notion))
+    for x in range(anthill.num_of_ants):
+        b1_notion, b2_notion = a_notion(x+2, days+5, anthill.num_of_ants+2, days+6)
+        sum_tips_list.append("=ROUNDDOWN({}*{};0)".format(b1_notion, b2_notion))
+    batch_cells(sum_tips_list, cell_list, wks)
+
+def SUM_total(anthill, days, wks):
+    '''
+    formula SUM total of the payment
+    '''
+    sum_total_list = []
+    a1_notion, a2_notion = a_notion(2,days+7,anthill.num_of_ants+1,days+7)
+    cell_list = wks.range("{}:{}".format(a1_notion, a2_notion))
+    for x in range(anthill.num_of_ants):
+        b1_notion, b2_notion = a_notion(x+2, days+4, x+2, days+6)
+        sum_total_list.append("=ROUNDDOWN({}+{};0)".format(b1_notion, b2_notion))
+    batch_cells(sum_total_list, cell_list, wks)
+
+def SUM_all(anthill, days, wks):
+    '''
+    SUM of hours for each day and colmuns during the month in the worksheet
+    '''
+    sum_all_list = []
+    a1_notion, a2_notion = a_notion(anthill.num_of_ants+2, 2, anthill.num_of_ants+2, days+7)
+    cell_list = wks.range("{}:{}".format(a1_notion, a2_notion))
+    # print(cell_list)
+    for x in range(1, days+7):
+        b1_notion, b2_notion = a_notion(2, x+1, anthill.num_of_ants+1, x+1)
+        sum_all_list.append("=SUM({}:{})".format(b1_notion, b2_notion))
+    batch_cells(sum_all_list, cell_list, wks)
+    wks.update_cell(2+anthill.num_of_ants, days+6, "0")
+
+def user_sheet_log(anthill_name, day_in, user_name, hours_delta, wks):
+    '''
+    writing the user time in the user cell with the day IN
+    '''
     if wks.cell(anthill_name[user_name].row, day_in+1).value == "":
-        wks.update_cell(anthill_name[user_name].row, day_in+1, "=ROUNDDOWN({};2)".format(hours))
+        wks.update_cell(anthill_name[user_name].row, day_in+1, "=ROUNDDOWN({};2)".format(hours_delta))
     else:
         cell_value = float(wks.cell(anthill_name[user_name].row, day_in+1).value)
-        total_hours = cell_value + hours
+        total_hours = cell_value + hours_delta
         wks.update_cell(anthill_name[user_name].row, day_in+1, "=ROUNDDOWN({};2)".format(total_hours))
 
-
-# labels for calculation row
-def labels_top(days_range, wks):
-    names_rows(wks)
-    dates_in_month(days_range, wks)
-    calculation_labels(days_range, wks)
-
-# formulas in SUMs
-def labels_SUM(days_range, wks):
-    SUM_hours(days_range, wks)
-    SUM_wage(days_range, wks)
-    SUM_wage99(days_range, wks)
-    SUM_tips_ratio(days_range, wks)
-    SUM_tips(days_range, wks)
-    SUM_total(days_range, wks)
-
-# botoms labels
-def labels_bottom(wks):
-    hours_label(wks)
-    SUM_day(days_range, wks)
-    SUM_labels(days_range, wks)
+def try_this_month():
```

Comment:
This code will crash when the exception happens.

wks is created only in the try branch, not the exception handling branch

Reply:
wks = auth_log(this_month)

is added to exception part